sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -80,13 +83,49 @@ case class CreateTableLikeCommand(
         s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
     }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    sourceTableDesc.tableType match {
+      case CatalogTableType.MANAGED | CatalogTableType.EXTERNAL | CatalogTableType.VIEW => // OK
+      case o => throw new AnalysisException(
+        s"CREATE TABLE LIKE is not allowed when the source table is ${o.name}")
+    }
+
+    if (DDLUtils.isDatasourceTable(sourceTableDesc) &&
+        sourceTableDesc.tableType != CatalogTableType.MANAGED) {
+      throw new AnalysisException(
+        "CREATE TABLE LIKE is not allowed when the source table is an external table created " +
+          "using the datasource API")
+    }
+
+    // For EXTERNAL_TABLE, the table properties has a particular field. To change it
+    // to a MANAGED_TABLE, we need to remove it; Otherwise, it will be EXTERNAL_TABLE,
+    // even if we set the tableType to MANAGED
+    // (metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1095-L1105)
+    // Table comment is stored as a table property. To clean it, we also should remove them.
+    val newTableProp =
+      sourceTableDesc.properties.filterKeys(key => key != "EXTERNAL" && key != "comment")
+    val newSerdeProp =
+      if (DDLUtils.isDatasourceTable(sourceTableDesc)) {
+        val newPath = catalog.defaultTablePath(targetTable)
+        sourceTableDesc.storage.properties ++ Map("path" -> newPath)
+      } else {
+        sourceTableDesc.storage.properties
+      }
+    val newTableDesc =
+      sourceTableDesc.copy(
```

Comment:
Great idea! Just do it in the lines: https://github.com/gatorsmile/spark/blob/6180e80c2bc69d23441b752f78f1ca94071dd0ba/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L64-L66